core-bundle/src/Filesystem/Dbafs/DbafsManager.php

```diff
@@ -0,0 +1,305 @@
+     * Each path is guaranteed to be only reported once, i.e. identical paths
+     * from DBAFs with a lower specificity will be ignored.
+     *
+     * @return \Generator<FilesystemItem>
+     */
+    public function listContents(string $path, bool $deep = false): \Generator
+    {
+        $covered = [];
+
+        foreach ($this->getDbafsForPath($path) as $prefix => $dbafs) {
+            foreach ($dbafs->getRecords(Path::makeRelative($path, $prefix), $deep) as $item) {
+                $itemPath = Path::join($prefix, $item->getPath());
+
+                if (\in_array($itemPath, $covered, true)) {
+                    continue;
+                }
+
+                $covered[] = $itemPath;
+
+                yield $item->withPath($itemPath);
+            }
+        }
+    }
+
+    /**
+     * Syncs all DBAFS that match all or parts of the given $paths.
+     */
+    public function sync(string ...$paths): ChangeSet
+    {
+        /** @var array<int, array{0: DbafsInterface, 1:string}> $dbafsDictionary */
+        $dbafsDictionary = [];
+
+        /** @var array<int, array<string>> $pathsForDbafs */
+        $pathsForDbafs = [];
+
+        foreach ($paths as $path) {
+            foreach ($this->getDbafsForPath(rtrim($path, '*')) as $prefix => $dbafs) {
+                $id = spl_object_id($dbafs);
+
+                $dbafsDictionary[$id] = [$dbafs, $prefix];
+                $pathsForDbafs[$id][] = Path::makeRelative($path, $prefix);
+            }
+        }
+
+        $changeSet = ChangeSet::createEmpty();
+
+        foreach ($pathsForDbafs as $id => $matchingPaths) {
+            [$dbafs, $prefix] = $dbafsDictionary[$id];
+
+            $changeSet = $changeSet->withOther($dbafs->sync(...$matchingPaths), $prefix);
+        }
+
+        return $changeSet;
+    }
+
+    /**
+     * @return \Generator<DbafsInterface>
+     */
+    private function getCandidatesForPrefix(string $prefix): \Generator
+    {
+        foreach ($this->dbafs as $dbafsPrefix => $dbafs) {
+            if (Path::isBasePath("/$prefix", "/$dbafsPrefix")) {
+                yield $dbafs;
+            }
+        }
+    }
+
+    /**
+     * @return \Generator<string, DbafsInterface>
+     */
+    private function getDbafsForPath(string $path): \Generator
+    {
+        foreach ($this->dbafs as $dbafsPrefix => $dbafs) {
+            if (Path::isBasePath("/$dbafsPrefix", "/$path")) {
+                yield $dbafsPrefix => $dbafs;
+            }
+        }
+    }
+
+    /**
+     * Make sure that all DBAFS with a more specific prefix are also supporting
+     * everything each less specific one does. For example a DBAFS with prefix
+     * 'files/media' must also support 'fileSize' if the DBAFS under 'files'
+     * does. It could, however, support additional properties like 'mimeType',
+     * even if the 'files' DBAFS does not.
+     */
+    private function validateTransitiveProperties(): void
+    {
+        $supports = [
```

Comment:
If we had a generalized `$dbafs->supportsFeature(DbafsInterface::FEATURE_LAST_MODIFIED)` and the like, we could easily add another feature in the future. Adding a constant to the interface wouldn't break it. Adding a method would.

Reply:
Good idea! :heart:  Implemented in e252d95b7e30d1cfe7918be08e1078709ccd608e.

I went with `getSupportedFeatures()` isntead of `supportsFeature()` because this way round they can be validated without knowing the complete list.